lib/jekyll/tags/post_url.rb

```diff
@@ -14,7 +14,8 @@ def initialize(name)
             "'#{name}' does not contain valid date and/or title."
         end
 
-        @name_regex = %r!^_posts/#{path}#{date}-#{slug}\.[^.]+|
+        escaped_slug = Regexp.escape(slug)
+        @name_regex = %r!^_posts/#{path}#{date}-#{escaped_slug}\.[^.]+|
           ^#{path}_posts/?#{date}-#{slug}\.[^.]+!x
```

Comment:
Thank you.  I had missed that one, and the updated pull request has both `slug`s escaped.